From af1b07238f0f12e6d3a5869d54c9dcae8902c67b Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Fri, 11 May 2018 23:43:21 +0800
Subject: [PATCH 3819/5242] MLK-18283-1 gpu: imx: imx8_dprc: Set prg stride
 bigger for blit engine

commit  339a0c5da967b20c8fb2d46fcaf2ecf2e50aee60 from
https://source.codeaurora.org/external/imx/linux-imx.git

Blit engine need arbitrary crop, but ULC can only lie on block boundary,
prg stride should be bigger than (width + align_margin) * bpp / 8

Signed-off-by: Yuchou Gan <yuchou.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index 431649f..7136721 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -372,7 +372,7 @@ void dprc_configure(struct dprc *dprc, unsigned int stream_id,
 	unsigned int prg_stride = width * info->cpp[0];
 	unsigned int bpp = 8 * info->cpp[0];
 	unsigned int preq;
-	unsigned int mt_w = 0, mt_h = 0;	/* w/h in a micro-tile */
+	unsigned int mt_w = 1, mt_h = 0;	/* w/h in a micro-tile */
 	u32 val;
 
 	if (WARN_ON(!dprc))
@@ -583,6 +583,9 @@ void dprc_configure(struct dprc *dprc, unsigned int stream_id,
 		}
 	}
 
+	if (dprc->is_blit_chan && dprc->devtype->has_fixup)
+		prg_stride = (width  + (x_offset % mt_w)) * info->cpp[0];
+
 	prg_configure(dprc->prgs[0], width, height, x_offset, y_offset,
 			prg_stride, bpp, baddr, format, modifier, start);
 	if (dprc->use_aux_prg)
-- 
1.7.9.5

